src/Extensions/Hosting.Services.Web/Middlewares/ObsoleteCorrelationHeadersMiddleware.cs

```diff
@@ -89,6 +95,25 @@ private void ExtractCorrelationFromRequest(HttpRequest request)
 				? value.FirstOrDefault(s => !string.IsNullOrWhiteSpace(s))
 				: null;
 
+		private static string? ExtractHeader(IHeaderDictionary headers, IEnumerable<string> names)
+		{
+			foreach (string name in names)
+			{
+				string? value = ExtractHeader(headers, name);
+				if (!string.IsNullOrWhiteSpace(value))
```

Comment:
You are checking twice for nullorwhitespace. You could add an out parameter to the function and return true when it is not null?